package/MDAnalysis/coordinates/TRR.py

```diff
@@ -1,123 +1,157 @@
-:attr:`~MDAnalysis.coordinates.xdrfile.TRR.Timestep.step` attribute
-and the mode coordinates are filling the
-:attr:`~MDAnalysis.coordinates.xdrfile.TRR.Timestep.positions` attribute of
-:class:`~MDAnalysis.coordinates.xdrfile.TRR.Timestep`::
-
-   # 'modes' is a mode object with M PCs, similar to a MxNx3 array
-   # 'xav' the average coordinates, a Nx3 array for N atoms
-
-   N = len(xav)   # number of atoms, i.e. number of coordinates
-
-   W = Writer('pca.trr', n_atoms=N)            # TRR writer
-   ts = MDAnalysis.coordinates.TRR.Timestep(N)  # TRR time step
-                                                # N of atoms is passed.
-   for frame,mode in enumerate(modes[4:16]):
-       ts.lmbda = -1
-
-       ts.frame = frame         # manually change the frame number
-       ts._frame = frame - 1
-
-       if frame<=1:
-          ts.positions = xav
-       else:
-          ts.positions = mode.scaledToNorm(1.).array*10   # nm to angstroms
-       if frame <= 1:
-          ts.time = frame-1
-       else:
-          ts.time = mode.frequency
-       W.write(ts)             # converts angstrom to nm for gmx
-
-    W.close()
-
-.. _MMTK: http://dirac.cnrs-orleans.fr/Manuals/MMTK/index.html
-
-.. _`recipe by Ramon Crehuet`: https://github.com/MDAnalysis/mdanalysis/issues/79
-
-Module reference
-----------------
-
-.. autoclass:: Timestep
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRReader
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRWriter
-   :members:
-   :inherited-members:
-"""
-
-from .xdrfile.TRR import TRRReader, TRRWriter, Timestep
+import errno
+
+from . import base
+from ..lib.formats.trr import TRRFile
+from ..lib.mdamath import triclinic_box, triclinic_vectors
+
+
+class TRRReader(base.Reader):
+    format = 'TRR'
+    units = {'time': 'ps', 'length': 'nm', 'velocity': 'nm/ps',
+             'force': 'kJ/(mol*nm)'}
+
+    def __init__(self, filename, convert_units=True, sub=None, **kwargs):
+        super(TRRReader, self).__init__(filename, convert_units=convert_units,
+                                        **kwargs)
+        self._trr = TRRFile(filename)
+
+        self.n_frames = len(self._trr)
+        self._sub = sub
+        trr_frame = self._trr.read()
+        try:
+            trr_frame_2 = self._trr.read()
+            time_2 = trr_frame_2.time
+        except:
+            time_2 = trr_frame.time
+
+        if sub is not None:
+            self.n_atoms = len(sub)
+        else:
+            self.n_atoms = self._trr.n_atoms
+        self._trr.seek(0)
+
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self._frame = 0
+        self.ts = self.frame_to_ts(trr_frame, self.ts)
+
+        self.ts.dt = time_2 - trr_frame.time
+        self.ts.dimensions = triclinic_box(*trr_frame.box)
+
+        if self.convert_units:
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+
+        self.rewind()
+
+    def frame_to_ts(self, frame, ts):
```

Comment:
I created a [notebook](https://gist.github.com/kain88-de/5487e9d9c8f7d0e4995b) benchmarking the different possible formats that are shipped with python.  You are welcome to run the benchmark yourself and report the results.

A remark to the marshal module. These files will likely not work across python versions or OSs.
Because they don't guarantee it they can optimize it for speed. Since we don't really exchange the offsets between machines I don't expect this to be a problem for us.

Also based on these results I would either go for marshal or json. Both are a better alternative then the pickle format.